### PR TITLE
feat: Add support for EC2 principal in assume worker role policy for China

### PR DESCRIPTION
**Pull request by GolubevV**
fix: EC2 principal for EKS worker role assume policy

# PR o'clock

## Description

Add logic to correctly handle EC2 principal service endpoint in China region for EKS Worker role assume policy.

### Checklist

- [x] CI tests are passing
- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

File: data.tf [hunk 1]
```diff
@@ -5,6 +5,7 @@ locals {
   worker_ami_name_filter_windows = (var.worker_ami_name_filter_windows != "" ?
     var.worker_ami_name_filter_windows : "Windows_Server-2019-English-Core-EKS_Optimized-${tonumber(var.cluster_version) >= 1.14 ? var.cluster_version : 1.14}-*"
   )
+  ec2_principal = (data.aws_partition.current.partition == "aws-cn") ? "ec2.amazonaws.com.cn" : "ec2.amazonaws.com"
```
Comment (dpiddockcmp) [inline data.tf:8]: `data.aws_partition` has a `dns_suffix` output. You can use it to simplify this logic.

Comment (GolubevV) [reply data.tf]: Done in new commit, together with fix in aws-auth for correct arn partition

File: aws_auth.tf [hunk 2]
```diff
@@ -2,9 +2,10 @@ data "aws_caller_identity" "current" {
 }
 
 locals {
+  arn_partition = data.aws_partition.current.partition
```
Comment (barryib) [inline aws_auth.tf:5]: We don't need this local, it's a simple assignment. I don't think it adds value.

Can you please replace `local.arn_partition` with `data.aws_partition.current.partition`

Comment (GolubevV) [reply aws_auth.tf]: Done

File: data.tf [hunk 3]
```diff
@@ -5,6 +5,7 @@ locals {
   worker_ami_name_filter_windows = (var.worker_ami_name_filter_windows != "" ?
     var.worker_ami_name_filter_windows : "Windows_Server-2019-English-Core-EKS_Optimized-${tonumber(var.cluster_version) >= 1.14 ? var.cluster_version : 1.14}-*"
   )
+  ec2_principal = "ec2.${data.aws_partition.current.dns_suffix}"
```
Comment (barryib) [inline data.tf:8]: Please. Same here.

Comment (barryib) [reply data.tf:8]: Ohhh it's ok for the ec2_principal.

**Comment by barryib** [comment]
Thanks @GolubevV for your contribution.

**Pull request merged**